docs/4.0-migration-guide/platform-entropy-gathering.md

```diff
@@ -0,0 +1,63 @@
+## Platform-specific Entropy Gathering
+
+### Disclaimer
+
+All the following sections refer to the case where entropy module uses default
+entropy sources, i.e. `MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES` is not defined.
+
+### Background
+
+Up to Mbed TLS 3.6 entropy module supported 3 entropy sources by default:
+
+- Platform
+  - Enabled by default it could be disabled through `MBEDTLS_NO_PLATFORM_ENTROPY`.
+  - It used platform-specific sources such as getrandom(), /dev/urandom or
+    BCryptGenRandom() to gather entropy data.
+  - Only useful when running on high level OSes, not in baremetal projects.
+
+- Hardware
+  - Disabled by default, it could be enabled through `MBEDTLS_ENTROPY_HARDWARE_ALT`.
+  - It allowed the user to define a custom function named `mbedtls_hardware_poll()`
+    to gather entropy data from some hardware source.
+  - Very useful for baremetal projects.
+  - Unfortunately the prototype for the function was in a private header file.
+
+- NV seed
+  - Disabled by default, it could be enabled through `MBEDTLS_ENTROPY_NV_SEED`.
+  - It allowed the user to store an entropy seed on a non-volatile (NV) memory
+    (ex: filesystem).
+  - This is crucial on systems that do not have a cryptographic entropy source
+    available through the previous 2 options.
+
+Up to Mbed TLS 3.6 all these 3 build symbols could be enabled independently.
+
+### Reasons to change
+
+Platform and Hardware entropy sources are most of the times mutually exclusive,
+so that the user will either use one or the other, but it's very unlikely to
+enable both. Therefore it's better to unify them under a single build symbol
+and to use a gathering function with the same name. Moreover this function must
+have the prototype in some public header so that the user can easily integrate
+it in their source code.
+
+### What changes in TF-PSA-Crypto 1.0?
+
+The following build symbols are removed:
+
+- `MBEDTLS_ENTROPY_HARDWARE_ALT`,
+- `MBEDTLS_ENTROPY_MIN_HARDWARE`,
+- `MBEDTLS_NO_PLATFORM_ENTROPY`.
+
+By default entropy module uses Platform entropy source to gather entropy data.
+However the user can enable `MBEDTLS_PLATFORM_GET_ENTROPY_ALT` to define their
+own custom implementation of the entropy gathering function and this will
+exclude the Platform one (alternate implementation).
+The prototype of this function can be found in `mbedtls/platform.h` and it's as
```

Comment:
This is describe a whole new feature, but that's not what is happening and that's not the goal of the migration guide. The target reader already has code that implements the old `mbedtls_hardware_poll`. So what we should explain here is, if you had an application that enabled `MBEDTLS_ENTROPY_HARDWARE_ALT`, how to change an implementation of `mbedtls_hardware_poll` into an implementation of `mbedtls_platform_get_entropy`.